### PR TITLE
OKTA-509008 : Enroll profile view parity spec

### PR DESCRIPTION
**Pull request by glenfannin-okta**
## PR Checklist

- [x] Have you verified the basic functionality for this change?
- [x] Did you add tests, as appropriate, following our [Automated Test guidelines](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/2676497890/Automated+Testing+in+the+Signin+Widget)?
- [x] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?
- [ ] Did you verify the change by running [downstream monolith artifact](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/102897979/Sign-in+Widget+Development#Sign-inWidgetDevelopment-Instructionstocreateandrunthedownstreamartifact(d16t))? (Provide link to build below)
- [ ] Does this PR include noticeable changes to the UI? (If yes, attach screenshots/video below)

### Issue:

- [OKTA-509008](https://oktainc.atlassian.net/browse/OKTA-509008)

### CODE REVIEW

File: playground/mocks/data/idp/idx/enroll-profile-with-password.json [hunk 1]
```diff
@@ -102,17 +102,17 @@
       ],
       "settings": {
         "complexity": {
-          "minLength": "NUMBER",
-          "minLowerCase": "NUMBER",
-          "minUpperCase": "NUMBER",
-          "minNumber": "NUMBER",
-          "minSymbol": "NUMBER",
-          "excludeUsername": "BOOLEAN",
```
Comment (glenfannin-okta) [inline playground/mocks/data/idp/idx/enroll-profile-with-password.json:110]: These values for this mock file are invalid, looked at others and they all have proper values (numeric or boolean). In v3, without updating this the requirements wont display.

File: src/v3/src/components/Form/renderers.tsx [hunk 2]
```diff
@@ -133,6 +133,26 @@ export default [
...
+  {
+    tester: ({
+      options: {
+        inputMeta: { options } = {} as Input,
+        format,
+        customOptions,
+      },
+    }: FieldElement) => (Array.isArray(customOptions) || Array.isArray(options)) && format === 'select',
+    renderer: Select,
+  },
```
Comment (glenfannin-okta) [inline src/v3/src/components/Form/renderers.tsx:155]: Had to move this up to ensure select/radio elements are rendered (the format option triggers these elements to be rendered).

File: test/testcafe/framework/page-objects/EnrollProfileViewPageObject.js [hunk 3]
```diff
@@ -1,5 +1,7 @@
 import BasePageObject from './BasePageObject';
 
+const requirementsSelector = '[data-se="password-authenticator--rules"]';
```
Comment (shawyu-okta) [inline test/testcafe/framework/page-objects/EnrollProfileViewPageObject.js:3]: feel free to inline this

File: src/v3/src/components/PasswordRequirements/PasswordRequirements.tsx [hunk 4]
```diff
@@ -97,7 +97,12 @@ const PasswordRequirements: UISchemaElementComponent<{
   return requirements?.length > 0 ? (
     <Box data-se="password-authenticator--rules">
       <Box marginBottom={2}>
-        <Text as="span">{header}</Text>
+        <Box
+          as="span"
+          className="password-authenticator--heading"
```
Comment (shawyu-okta) [inline src/v3/src/components/PasswordRequirements/PasswordRequirements.tsx:102]: Could the header text be picked up with `getByText` or ever better, `getByRole('heading'`? I noticed a lot of existing `EnrollProfileView_spec` code has very explicit CSS-type selector strings embedded in it, which we should make an effort to rewrite in the same spirit if possible.

File: test/testcafe/framework/page-objects/EnrollProfileViewPageObject.js (hunk 3 above)
Comment (glenfannin-okta) [reply test/testcafe/framework/page-objects/EnrollProfileViewPageObject.js]: I ultimately got rid of this and used the `getByText` function instead based on your other feedback. 😄

File: src/v3/src/components/PasswordRequirements/PasswordRequirements.tsx (hunk 4 above)
Comment (glenfannin-okta) [reply src/v3/src/components/PasswordRequirements/PasswordRequirements.tsx:102]: I updated the test to use `this.form.getByText(text).exists` and passing the expected text into this function.

**Review by shawyu-okta — APPROVED** [review]

**Pull request merged**